desktop/libs/indexer/src/indexer/api3.py

```diff
@@ -456,7 +456,7 @@ def importer_submit(request):
       stream = request.fs.open(path)
       file_encoding = check_encoding(stream.read(10000))
 
-  if destination['ouputFormat'] in ('database', 'table'):
+  if destination['ouputFormat'] in ('database', 'table') and hasattr(request, 'fs'):
```

Comment:
Why `fs` is not set in the middleware?

Reply:
If related to small fill and no FS configured, is `netnormpath` available via a class method?

Reply:
can't tell much what subrata is facing that must be related to `request.fs` thing as logs are

`[15/Jul/2021 15:00:25 -0700] access       INFO     10.96.87.60, 127.0.0.1,127.0.0.1 csso_nandi - "POST /indexer/api/importer/submit HTTP/1.1" returned in 2ms 200 77

[15/Jul/2021 15:00:25 -0700] decorators   ERROR    Error running importer_submit

Traceback (most recent call last):

  File "/opt/hive/desktop/libs/notebook/src/notebook/decorators.py", line 119, in wrapper

    return f(*args, **kwargs)

  File "/opt/hive/desktop/libs/indexer/src/indexer/api3.py", line 460, in importer_submit

    if destination['nonDefaultLocation'] else destination['nonDefaultLocation']

AttributeError: 'NoneType' object has no attribute 'netnormpath'

 `

Reply:
Read more about the FS middleware and if netnormpath can be accessed without an instance if it is a classmethod..

Reply:
@roamin So here's what I found -> **so I am able to reproduce this error locally also when hdfs and s3 are disabled.**  
and netnormpath is not a classmethod.

so in this PR some improvements needed or good to go?

Reply:
looks like it is related to FS (giving an error if not configured) and not related to hive or impala.

Reply:
so so request.fs is none it is giving error.

Reply:
???